ingress/elb/elb.go

```diff
@@ -65,18 +75,33 @@ func (e *elb) Attach(frontend types.FrontendInput) (int, error) {
 		return 0, err
 	}
 
+	// Save these now so we can always know what we might have done
+	// up until this point we have only read data
+	e.elbs = clusterFrontEnds
+	e.instanceID = instance
+	registered := 0
+
+	if len(clusterFrontEnds) != e.expectedFrontends {
+		log.Warnf("Expected to find %d elb frontends, only found %d", e.expectedFrontends, len(clusterFrontEnds))
+	}
+
 	for _, frontend := range clusterFrontEnds {
-		// TODO deal with error
-		e.awsElb.RegisterInstancesWithLoadBalancer(&aws_elb.RegisterInstancesWithLoadBalancerInput{
+		log.Infof("Registering instance %s with elb %", instance, frontend)
+		_, err = e.awsElb.RegisterInstancesWithLoadBalancer(&aws_elb.RegisterInstancesWithLoadBalancerInput{
 			Instances: []*aws_elb.Instance{
 				&aws_elb.Instance{
 					InstanceId: aws.String(instance),
 				}},
 			LoadBalancerName: aws.String(frontend),
 		})
 
+		if err != nil {
+			return registered, fmt.Errorf("unable to register instance %s with elb %s: %v", instance, frontend, err)
```

Comment:
Actually the controller fails to start if any of its Start() methods fail so not worth doing this unless we change that